src/hooks/useFetchProducts.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+import { productService } from "../services/products/product.service";
+import { Product } from "../interfaces/products/product.interface";
+
+export const useFetchProducts = () => {
+  const [products, setProducts] = useState<Product[] | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    const fetchAllProducts = async () => {
```

Comment:
Nice that you changed here to have a custom hook. If you have time you could do this for the other logic as well.